api/middleware/apd.test.js

```diff
@@ -0,0 +1,206 @@
+const tap = require('tap');
+const sandbox = require('sinon').createSandbox();
+
+const middleware = require('./apd');
+
+const OtherModel = {
+  where: sandbox.stub(),
+  fetch: sandbox.stub()
+};
+
+const res = {
+  status: sandbox.stub(),
+  send: sandbox.stub(),
+  end: sandbox.stub()
+};
+const next = sandbox.spy();
+
+tap.beforeEach(done => {
+  sandbox.resetBehavior();
+  sandbox.resetHistory();
+
+  res.status.returns(res);
+  res.send.returns(res);
+  res.end.returns(res);
+
+  done();
+});
+
+tap.test('APD-related middleware', async middlewareTests => {
+  middlewareTests.test('load apd', async loadApdTests => {
+    loadApdTests.skip('from an APD model', async () => {
+      // There's an in-built dependency on the real APD model,
+      // so we need to figure out how to either mock that in
+      // such a way that we fool the code, OR we need to find
```

Comment:
would it be easier if we checked and mocked against an instance method on the APD model, i.e., `if model.foo == ApdModel.foo` ?

Reply:
I poked around on the bookshelf object to try to find something that would uniquely identify the model (e.g., wouldn't it be cool if `model.name` or `model.tableName` existed?), but I couldn't find anything that would be unique to the APD model.

But we could inject the "control" as well, I suppose:

```js
module.exports.loadApd = (model = ApdModel, idParam = 'id', ApdModel = ApdModel) => {
...
}
```

It seems hacky but a lot of the model injection we're doing elsewhere is ultimately just to enable testing so maybe not?  🤷‍♂️

Reply:
yeah, `model.tableName` would have been perfect, alas... hmm yeah it does seem a little hacky but i'm fine with adding that param for now